src/Domain/RetailerDemListCreator.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace Linn.DemStock.Domain
+{
+    using Linn.Common.Persistence;
+    using Linn.DemStock.Domain.Exceptions;
+    using Linn.DemStock.Domain.Repositories;
+    using Linn.DemStock.Domain.RetailerDemListActivities;
+
+    public class RetailerDemListCreator : ActivityEntity<RetailerDemListActivity>
```

Comment:
This class doesn't look right...if it's an Entity then it it's a persisted domain object, which means it shouldn't have access to transactions or repositories. If it's a Service to handle the creation of Retailer Dem Lists, it should be at the Facade level and shouldn't be extending ActivityEntity. This looks like a halfway house between the two?